Resources/doc/index.rst

```diff
@@ -0,0 +1,160 @@
+        pattern:  /productDocument/{document}/ # This pattern is ignored and required for compatibility with Symfony.
+        defaults: { _controller: ONGRDemoBundle:Product:document }
+    
+    # Actual route for accessing by document id (not SEO).
+    ongr_product_show:
+        pattern:  /product/{id}
+        defaults: { _controller: ONGRDemoBundle:Product:show }
+        requirements:
+            page:  \d+
+..
+
+.. code-block:: yaml
+
+    ongr_router:
+        es_manager: default
+        seo_routes:
+            product:
+                _route: ongr_product
+                _controller: ONGRDemoBundle:Product:document
+                _default_route: ongr_product_show
+                _id_param: documentId
+..
+
+Basic parameters for each routing configuration are:
+* `_route` - Main route for matching and generating SEO URLs.
+* `_controller` - controller this route will use.
+* `_default_route` - route which will be used if generating URL using RouterBundle fails. E. g. document has no SEO URLs assigned.
+* `_id_param` - id field to be passed to `_default_route`.
+
+Router manager is ElasticsearchBundle manager name which will be used to get documents.
+
+Generating URL
+~~~~~~~~~~~~~~
+
+Use Symfony's default generator to produce links to specific documents. For example:
+
+.. code-block:: twig
+
+    <a href="{{ path('ongr_product', {'document': product, '_seo_key': 'summer'} }}">My Product</a>
+..
+
+Such template will generate SEO link to the document. Function `path` is just an example. You can also choose `url` or plain `$this->get('router')->generate(...)` generator.
+
+SEO key
+~~~~~~~
+
+Conroller action, document URLs and route parameters all have SEO key variable. Since document can have multiple SEO URLs, e. g. for multiple categories or sections, one can differentiate URLs by SEO key. It is optional, but key as a simple string can be passed to the URL generator (like in the example above). If the key is omitted, first document's URL is used.
+
+Alternative action
+~~~~~~~~~~~~~~~~~~
+
+It is possible for the document to not have any SEO URLs defined. Therefore, it is recommended to include such action in the controller:
+
+.. code-block:: php
+
+    public function productAction($productId) {
+        $product = $this->get('es.manager')->getRepository('product')->find($productId);
+    
+        if ($productId === null) {
+            throw $this->createNotFoundException();
+        }
+    
+        return $this->render(
+            // Your template.
+            $this->getProductTemplate($product),
+            [
+                'product' => $product
+            ]
+        );
+    }
+..
+
+SEO generator will use this action and it's associated route to produce URL from `ongr_product_show` or similar route defined in `_default_route`. This route and action will use document ID as it's fallback argument, not SEO URL.
+
+Setup
+~~~~~
+
+Setup documentation for the Router bundle is available `here <setup.rst>`_.
```

Comment:
You can leave just ``... available in `<setup.rst>`_ page`` and it will insert page title automatically.